Sources/SwiftGRPCNIO/GRPCChannelHandler.swift

```diff
@@ -28,30 +28,32 @@ public protocol CallHandlerProvider: class {
 /// for an `GRPCCallHandler` object. That object is then forwarded the individual gRPC messages.
 public final class GRPCChannelHandler {
   private let servicesByName: [String: CallHandlerProvider]
+  private weak var errorDelegate: ServerErrorDelegate?
 
-  public init(servicesByName: [String: CallHandlerProvider]) {
+  public init(servicesByName: [String: CallHandlerProvider], errorDelegate: ServerErrorDelegate? = nil) {
     self.servicesByName = servicesByName
+    self.errorDelegate = errorDelegate
   }
 }
 
 extension GRPCChannelHandler: ChannelInboundHandler {
   public typealias InboundIn = RawGRPCServerRequestPart
   public typealias OutboundOut = RawGRPCServerResponsePart
-  
+
+  public func errorCaught(ctx: ChannelHandlerContext, error: Error) {
```

Comment:
Mild code duplication from `BaseCallHandler` here, but not sure whether we can (or should) do anything about that.

Reply:
I don't think there's a lot we can (reasonably) do!